File: spring-data-requery-test-java/src/main/java/org/springframework/data/requery/domain/sample/AbstractMailSender.java
package org.springframework.data.requery.domain.sample;

import io.requery.*;
import org.jetbrains.annotations.NotNull;
import org.springframework.data.requery.domain.AbstractPersistable;
import org.springframework.data.requery.domain.ToStringBuilder;

import java.util.Objects;

/**
 * AbstractMailSender
 *
 * @author debop@coupang.com
 * @since 18. 6. 25
 */
@Entity
public abstract class AbstractMailSender extends AbstractPersistable<Long> {

    @Key
    @Generated
    protected Long id;

    protected String name;

    @ForeignKey
    @ManyToOne
    protected AbstractMailUser mailUser;

    public AbstractMailSender() {}

    public AbstractMailSender(String name) {
        this.name = name;
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(name);
    }

    @Override
    protected @NotNull ToStringBuilder buildStringHelper() {
        return super.buildStringHelper()
            .add("name", name);
    }

    private static final long serialVersionUID = -6924337437200476666L;
}

File: src/test/kotlin/com/coupang/springframework/data/requery/domain/sample/AbstractMailSender.kt
package com.coupang.springframework.data.requery.domain.sample

import com.coupang.springframework.data.requery.domain.AbstractPersistable
import io.requery.Entity
import io.requery.Generated
import io.requery.Key
import io.requery.Lazy
import io.requery.ManyToOne

/**
 * com.coupang.springframework.data.requery.domain.sample.AbstractMailSender
 * @author debop
 * @since 18. 5. 30
 */
@Entity
abstract class AbstractMailSender: AbstractPersistable<Long>() {

    @get:Key
    @get:Generated
    abstract override val id: Long?

    abstract var name: String?

    @get:ManyToOne
    @get:Lazy
    abstract var mailUser: AbstractMailUser?

}